src/lib/compress_hw.c

```diff
@@ -119,6 +119,52 @@ static bool _is_codec_type_supported(int fd, struct snd_codec *codec)
 	return found;
 }
 
+static int compress_hw_get_supported_codecs_by_name(const char *name, unsigned int flags, void *codecs, int size)
+{
+	struct snd_compr_caps caps;
+	unsigned int *codecs_buf = (unsigned int *)codecs;
+	unsigned int card, device;
+	unsigned int dev_flag;
+	unsigned int num_codecs, i;
+	int fd, ret;
+	char fn[256];
+
+	if (sscanf(&name[3], "%u,%u", &card, &device) != 2)
+		return -EINVAL;
+
+	snprintf(fn, sizeof(fn), "/dev/snd/comprC%uD%u", card, device);
+
+	if (flags & COMPRESS_OUT)
+		dev_flag = O_RDONLY;
+	else
+		dev_flag = O_WRONLY;
+
+	fd = open(fn, dev_flag);
+	if (fd < 0)
+		return oops(&bad_compress, errno, "cannot open device '%s'", fn);
+
+	if (ioctl(fd, SNDRV_COMPRESS_GET_CAPS, &caps)) {
+		oops(&bad_compress, errno, "cannot get device caps");
+		close(fd);
+		return -EINVAL;
+	}
+
+	if (caps.num_codecs > size) {
+		oops(&bad_compress, EINVAL, "Input is not big enough to hold list of available codecs");
+		close(fd);
+		return -EINVAL;
```

Comment:
that's too late, there's already a risk of buffer overflow.

Reply:
why do you make something as resolved when there's no sign that the buffer overflow issue was resolved?